examples/basic_cog.py

```diff
@@ -0,0 +1,116 @@
+
+
+# This defines a class that contains commands in a "Miscellaneous" category.
+# The library calls this concept a Cog and they must inherit from `commands.Cog`.
+# Cogs are useful for grouping commands and having them share state.
+# This class will group commands in a "Misc" category in the default help command.
+# The `name` keyword argument passed allows us to set the name of the Cog.
+# If a name is not given then it will default to the name of the class.
+# Read more here: https://discordpy.readthedocs.io/en/latest/ext/commands/cogs.html.
+
+
+class Miscellaneous(commands.Cog, name='Misc'):
+    """
+    A simple Miscellaneous cog.
+    """
+
+    def __init__(self, bot):
+
+        # We pass in the bot parameter when initializing the class, and make it an instance attribute.
+        # This lets us access the bot parameter throughout our methods.
+        self.bot = bot
+        # You can also add other attributes to the class.
+        self.afk = {}
+
+    # Cogs also have the capability to define checks that apply to every command in the Cog.
+    # They work similarly to regular checks elsewhere in the framework.
+    # ref: https://discordpy.readthedocs.io/en/latest/ext/commands/commands.html#checks
+    async def cog_check(self, ctx):
+        # Check if the command was invoked within a Guild channel.
+        return ctx.guild is not None
+
+    # Commands within a Cog are defined using the commands.command decorator
+    # instead of the bot.command decorator.
+    # They must also pass a `self` parameter since it's within a class.
+    @commands.command()
+    async def hello(self, ctx):
+        """Adds two numbers together"""
+        await ctx.send("Hello am I an example bot!")
+
+    # A command group within a Cog.
+    # The `invoke_without_command` flag for the group allows us to use the parent of the group as a command.
+    @commands.group(invoke_without_command=True)
+    async def afk(self, ctx, *, reason: str):
+        """
+        Set your afk status
+        """
+        u_id = ctx.author.id
+        # Simply check if they exist in our afk dictionary aldready.
+        # If they exist then tell them they are afk, or let them set a new afk status.
+        rating = self.afk.get(u_id)
+        if not rating:
+            self.afk[u_id] = reason
+            await ctx.send(f"{ctx.author.mention} I have set your afk status as `{reason}`.")
+        else:
+            await ctx.send(f"{ctx.author.mdention} you are aldready afk.")
+
+    # Subcommand for the group
+    @afk.command(name='user')
+    async def _user(self, ctx, *, user: discord.User = None):
+        """Check a user's afk status"""
+        if not user:
+            return await ctx.send("I need a user to check the afk status")
+        afk = self.afk.get(user.id)
+        if afk:
+            return await ctx.send(f"{user.name} is afk with reason `{afk}`")
+        return await ctx.send(f'{user.name} is currently not afk')
```

Comment:
try-except works better for this case. Also don't use f-strings, use allowed_mentions, etc. just like above.